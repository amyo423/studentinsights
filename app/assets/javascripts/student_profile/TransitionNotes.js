import React from 'react';
import PropTypes from 'prop-types';
import SectionHeading from '../components/SectionHeading';
import _ from 'lodash';

const styles = {
  textarea: {
    marginTop: 20,
    fontSize: 14,
    border: '1px solid #eee',
    width: '100%' //overriding strange global CSS, should cleanup
  }
};

const notePrompts = `What are this student's strengths?
——————————


What is this student's involvement in the school community like?
——————————


How does this student relate to their peers?
——————————


Who is the student's primary guardian?
——————————


Any additional comments or good things to know about this student?
——————————


`;

const restrictedNotePrompts = `Is this student receiving Social Services and if so, what is the name and contact info of their social worker?
——————————


Is this student receiving mental health supports?
——————————
  `;

class TransitionNotes extends React.Component {

  constructor(props) {
    super(props);

    const transitionNotes = props.transitionNotes;
    const regularNote = _.find(transitionNotes, {is_restricted: false});
    const restrictedNote = _.find(transitionNotes, {is_restricted: true});

    this.state = {
      noteText: (regularNote ? regularNote.text : notePrompts),
      restrictedNoteText: (restrictedNote ? restrictedNote.text : restrictedNotePrompts),
      regularNoteId: (regularNote ? regularNote.id : null),
      restrictedNoteId: (restrictedNote ? restrictedNote.id : null)
    };

    this.onClickSave = this.onClickSave.bind(this);
    this.onClickSaveRestricted = this.onClickSaveRestricted.bind(this);
  }

  buttonText() {
    const {requestState} = this.props;

    if (requestState === 'pending') return 'Saving ...';

    if (requestState === 'error') return 'Error ...';

    return 'Save Note';
  }

  buttonTextRestricted() {
    const {requestStateRestricted} = this.props;

    if (requestStateRestricted === 'pending') return 'Saving ...';

    if (requestStateRestricted === 'error') return 'Error ...';

    return 'Save Note';
  }

  onClickSave() {
    const params = {
      is_restricted: false,
      text: this.state.noteText
    };

    if (this.state.regularNoteId) {
      _.merge(params, {id: this.state.regularNoteId});
    }

    this.props.onSave(params);
  }

  onClickSaveRestricted() {
    const params = {
      is_restricted: true,
      text: this.state.restrictedNoteText
    };

    if (this.state.restrictedNoteId) {
      _.merge(params, {id: this.state.restrictedNoteId});
    }

    this.props.onSave(params);
  }

  render() {
    const {noteText, restrictedNoteText} = this.state;

    return (
      <div style={{display: 'flex'}}>
        <div style={{flex: 1, margin: 30}}>
          <SectionHeading>
            High School Transition Note
          </SectionHeading>
          <textarea
            rows={10}
            style={styles.textarea}
            value={noteText}
            onChange={(e) => this.setState({noteText: e.target.value})} />
          <button onClick={this.onClickSave} className="btn save">
            {this.buttonText()}
          </button>
        </div>
        <div style={{flex: 1, margin: 30}}>
          <SectionHeading>
            High School Transition Note (Restricted)
          </SectionHeading>
          <textarea
            rows={10}
            style={styles.textarea}
            value={restrictedNoteText}
            onChange={(e) => this.setState({restrictedNoteText: e.target.value})}/>
          <button onClick={this.onClickSaveRestricted} className="btn save">
            {this.buttonTextRestricted()}
          </button>
        </div>
      </div>
    );
  }

}

TransitionNotes.propTypes = {
  readOnly: PropTypes.bool.isRequired,
  onSave: PropTypes.func.isRequired,
  transitionNotes: PropTypes.array.isRequired,
  requestState: PropTypes.string,              // can be null if no request
  requestStateRestricted: PropTypes.string     // can be null if no request
};

export default TransitionNotes;

