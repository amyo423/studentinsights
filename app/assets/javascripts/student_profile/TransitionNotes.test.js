import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TransitionNotes from './TransitionNotes';

function testProps(props) {
  return {
    readOnly: false,
    onSave: jest.fn(),
    defaultTransitionNotes: [],
    requestState: null,
    requestStateRestricted: null,
    ...props
  };
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  ReactDOM.render(<TransitionNotes {...testProps()} />, el);
  expect(el.innerHTML).toContain('High School Transition Note');
  expect(el.innerHTML).toContain('High School Transition Note (Restricted)');
});

it('snapshots view', () => {
  const tree = renderer
    .create(<TransitionNotes {...testProps()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});