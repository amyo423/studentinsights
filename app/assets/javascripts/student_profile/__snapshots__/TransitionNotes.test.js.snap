// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots view 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    style={
      Object {
        "flex": 1,
        "margin": 30,
      }
    }
  >
    <div
      className="SectionHeading"
      style={
        Object {
          "borderBottom": "1px solid #333",
          "padding": 10,
        }
      }
    >
      <h4
        style={
          Object {
            "color": "black",
            "display": "inline",
          }
        }
      >
        High School Transition Note
      </h4>
    </div>
    <textarea
      onChange={[Function]}
      readOnly={undefined}
      rows={10}
      style={
        Object {
          "border": "4px solid rgba(153,117,185, 0.4)",
          "fontSize": 14,
          "marginTop": 20,
          "width": "100%",
        }
      }
      value="What are this student's strengths?

What is this student's involvement in the school community like?

How does this student relate to their peers?

Who is the student's primary guardian?

Any additional comments or good things to know about this student?"
    />
    <button
      className="btn save "
      disabled={false}
      onClick={[Function]}
    >
      Save Note
    </button>
  </div>
  <div
    style={
      Object {
        "flex": 1,
        "margin": 30,
      }
    }
  >
    <div
      className="SectionHeading"
      style={
        Object {
          "borderBottom": "1px solid #333",
          "padding": 10,
        }
      }
    >
      <h4
        style={
          Object {
            "color": "black",
            "display": "inline",
          }
        }
      >
        High School Transition Note (Restricted)
      </h4>
    </div>
    <textarea
      onChange={[Function]}
      readOnly={undefined}
      rows={10}
      style={
        Object {
          "border": "4px solid rgba(153,117,185, 0.4)",
          "fontSize": 14,
          "marginTop": 20,
          "width": "100%",
        }
      }
      value="Is this student receiving Social Services and if so, what is the name and contact info of their social worker?

Is this student receiving mental health supports?"
    />
    <button
      className="btn save "
      disabled={false}
      onClick={[Function]}
    >
      Save Note
    </button>
  </div>
</div>
`;
